packages/schematics/src/cypress/index.spec.ts

```diff
@@ -0,0 +1,282 @@
+                root: 'projects/app-e2e/',
+                projectType: 'application',
+                prefix: '',
+                architect: {
+                  e2e: {}
+                }
+              }
+            }
+          })
+        );
+      });
+
+      it('should set correct outDir compiler option in tsconfig.e2e.json', () => {
+        const treeAfter = runner.runSchematic(
+          'cypress',
+          parameters,
+          treeBefore
+        );
+        const tsConfigE2eJson = JSON.parse(
+          treeAfter.readContent('/projects/app-e2e/tsconfig.e2e.json')
+        );
+        expect(tsConfigE2eJson.compilerOptions.outDir).toEqual(
+          '../../dist/out-tsc/apps/my-app-e2e/src'
+        );
+      });
+
+      it('should set correct extends option in tsconfig.json', () => {
+        const treeAfter = runner.runSchematic(
+          'cypress',
+          parameters,
+          treeBefore
+        );
+        const tsConfigJson = JSON.parse(
+          treeAfter.readContent('/projects/app-e2e/tsconfig.json')
+        );
+        expect(tsConfigJson.extends).toEqual('../../tsconfig.json');
+      });
+
+      it('should use correct directories in cypress.json', () => {
+        const treeAfter = runner.runSchematic(
+          'cypress',
+          parameters,
+          treeBefore
+        );
+        const cypressJson = JSON.parse(
+          treeAfter.readContent('/projects/app-e2e/cypress.json')
+        );
+        expect(cypressJson).toEqual(
+          expect.objectContaining({
+            fileServerFolder: '../../dist/out-tsc/apps/my-app-e2e',
+            fixturesFolder: '../../dist/out-tsc/apps/my-app-e2e/src/fixtures',
+            integrationFolder:
+              '../../dist/out-tsc/apps/my-app-e2e/src/integration',
+            pluginsFile:
+              '../../dist/out-tsc/apps/my-app-e2e/src/plugins/index.js',
+            videosFolder: '../../dist/out-tsc/apps/my-app-e2e/videos',
+            screenshotsFolder: '../../dist/out-tsc/apps/my-app-e2e/screenshots'
+          })
+        );
+      });
+    });
+
+    it('should set an npm script for cypress open', () => {
+      treeBefore.create('e2e/.keep', '');
+      treeBefore.create(
+        'angular.json',
+        JSON.stringify({
+          projects: {
+            'my-app': {
+              projectType: 'application',
+              architect: {}
+            },
+            'my-app-e2e': {
+              root: 'e2e/',
+              projectType: 'application',
+              architect: { e2e: { options: {} } }
+            }
+          }
+        })
+      );
+
+      const parameters = {
+        ...defaultParameters,
+        app: 'my-app',
+        overwrite: true
+      };
+
+      const tree = runner.runSchematic('cypress', parameters, treeBefore);
+
+      const packageJson: PackageJsonSchema = JSON.parse(
+        tree.readContent('package.json')
+      );
+
+      expect(packageJson.scripts).toEqual({ 'cy:open': 'cypress open' });
```

Comment:
Can you add the script `cypress run`?
This command will be run by the CI server.

Reply:
I wasn't sure if we really need that, because you can run `ng e2e [project-name] --headless`

Reply:
`cypress run` script would be used like this: `npm run cy:run -- --project=[project-name]`

Reply:
Added a general `cy:run` script.

Reply:
But another advantage of `ng e2e` is, that it runs tests of all e2e projects in workspace, if there are more than one.